### PR TITLE
sinf tanf acosf asinf atanf expm1f

### LINKED ISSUE #34: implement sinf
Source: https://git.musl-libc.org/cgit/musl/tree/src/math/sinf.c

### LINKED ISSUE #39: implement acosf
Source: https://git.musl-libc.org/cgit/musl/tree/src/math/acosf.c

### LINKED ISSUE #40: implement asinf
Source: https://git.musl-libc.org/cgit/musl/tree/src/math/asinf.c

### LINKED ISSUE #42: implement atanf
Source: https://git.musl-libc.org/cgit/musl/tree/src/math/atanf.c

### LINKED ISSUE #45: implement expm1f
Source: https://git.musl-libc.org/cgit/musl/tree/src/math/expm1f.c

### LINKED ISSUE #51: implement tanf
Source: https://git.musl-libc.org/cgit/musl/tree/src/math/tanf.c

### PR DESCRIPTION
**Pull request by burrbull**
closes #39 
closes #40 
closes #42 
closes #45 
closes #34 
closes #51

### CODE REVIEW

**Comment by japaric** [comment]
Thanks @burrbull 

bors r+

**Comment by japaric** [comment]
sinhf has some failures:

https://travis-ci.org/japaric/libm/jobs/403962524#L762
https://travis-ci.org/japaric/libm/jobs/403962526#L762
https://travis-ci.org/japaric/libm/jobs/403962521#L762

Also, cargo-fmt failed:

https://travis-ci.org/japaric/libm/jobs/403962532#L474

**Comment by japaric** [comment]
> sinhf has some failures:

@burrbull you can remove the sinhf implementation for now, then we can land the other implementations in this PR.

**Comment by japaric** [comment]
> Can I see what input value leads to panic in sinhf ?

From the Travis links:

> 	thread 'sinhf' panicked at 'input: 3266435980, output: 4280104378, expected: 4280108624', tests/sinhf.rs:25:37

> 	thread 'sinhf' panicked at 'input: 1119011708, output: 2138524592, expected: 2138531271', tests/sinhf.rs:25:37

> 	thread 'sinhf' panicked at 'input: 3266487942, output: 4285116960, expected: 4285123271', tests/sinhf.rs:25:37

The values are off by thousands of ULP though.

**Comment by burrbull** [comment]
Is 3266435980 float?
On these values I get `inf` result.
Same with std .sinh()

How to delete commit (file) from PR?
Close and make new?

**Comment by japaric** [comment]
> Is 3266435980 float?

No, that's the integer representation. You need to call `f32::from_bits` before passing that to sinhf.

> How to delete commit (file) from PR?

`git rm path/to/file`, locally; then `git commit`; and then `git push origin master`

File: src/lib.rs [hunk 1]
```diff
@@ -300,19 +283,16 @@ impl F32Ext for f32 {
         log1pf(self)
     }
 
-    #[cfg(todo)]
```
Comment (japaric) [inline src/lib.rs]: needs to be undone because sinhf has been removed

File: src/math/mod.rs [hunk 2]
```diff
@@ -39,12 +42,17 @@ mod round;
 mod roundf;
 mod scalbn;
 mod scalbnf;
+mod sinf;
+mod sinhf;
```
Comment (japaric) [inline src/math/mod.rs]: needs to be removed because sinhf has been removed

File: src/math/mod.rs [hunk 3]
```diff
@@ -78,14 +87,23 @@ pub use self::round::round;
 pub use self::roundf::roundf;
 pub use self::scalbn::scalbn;
 pub use self::scalbnf::scalbnf;
+pub use self::sinf::sinf;
+pub use self::sinhf::sinhf;
```
Comment (japaric) [inline src/math/mod.rs]: needs to be removed because sinhf has been removed

File: test-generator/src/main.rs [hunk 4]
```diff
@@ -651,26 +651,29 @@ fn main() -> Result<(), Box<Error>> {
...
     log1pf,
     log2f,
     logf,
     roundf,
-    // sinf,
-    // sinhf,
-    // tanf,
-    // tanhf,
+    sinf,
+    sinhf,
```
Comment (japaric) [inline test-generator/src/main.rs]: needs to be undone because sinhf has been removed

**Pull request closed without merging**